File: Medium/0003-longest-substring/Solution.java
import java.util.HashSet;
import java.uitl.set;

Class Solution {
    public int LengthOfLongestSubstring(String s) {
        set<Character> charSet = new HashSet<>();
        int left = 0, maxLength = 0;

        for (int right = 0; right < s.Length(); right++) {

            while (charSet.contains(s.charAt(right))) {
                charSet.remove(s.charAt(left));
                left++;
            }

            charSet.add(s.charAt(right));
            maxLength = Math.max(maxLength, right - left + 1);
        }
        return maxLength;
    }
}

File: Medium/0003-longest-substring/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        last_index = {}
        left = 0
        max_len = 0

        for right, c in enumerate(s):
            if c in last_index and last_index[c] >= left:
                left = last_index[c] + 1
            last_index[c] = right
            max_len = max(max_len, right - left + 1)

        return max_len